cfme/tests/test_rest.py

```diff
@@ -374,13 +374,485 @@ def test_vm_add_lifecycle_event(request, setup_a_provider, rest_api, vm, from_de
+
+@pytest.fixture(scope="module")
+def added_policies(request, policies_data, rest_api):
+    assert "add" in rest_api.collections.policies.action
+
+    for policy in policies_data:
+        rest_api.collections.policies.action.add(policy)
+        wait_for(
+            lambda: rest_api.collections.policies.find_by(name=policy.get("description")),
+            num_sec=180,
+            delay=10,
+        )
+
+    def fin():
+            policies = [policy.id for policy in rest_api.collections.policies]
+            rest_api.collections.policies.delete(policies)
+    request.addfinalizer(fin)
+
+    return [policy.id for policy in rest_api.collections.policies]
+
+
+def test_add_delete_policy_profiles(policy_profiles_data, rest_api):
+    """Tests creating and deleting policy_profiles
+
+    Prerequisities:
+        * An appliance with ``/api`` available.
+
+    Steps:
+        * POST /api/policy_profiles (method ``add``)
+        * DELETE /api/policy_profiles/<id> -> delete only one policy profile
+        * DELETE /api/policy_profiles <- Insert a JSON with list of dicts containing ``href``s to
+            the policy profile
+        * Repeat the DELETE query -> now it should return an ``ActiveRecord::RecordNotFound``.
+
+    Metadata:
+        test_flag: rest
+    """
+    assert ("add" in rest_api.collections.policy_profiles.action and
+        "delete" in rest_api.collections.policy_profiles.action)
+
+    for policy_profile in policy_profiles_data:
+        rest_api.collections.policy_profiles.action.add(policy_profile)
+
+        wait_for(
+            lambda: rest_api.collections.policy_profiles.find_by(
+                name=policy_profile.get("description")),
+            num_sec=180,
+            delay=10,
+        )
+
+    delete_policy_profile = rest_api.collections.policy_profiles.find_by(
+        policy_profiles_data[0].get('description'))
+    delete_policy_profile.action.delete()
+    wait_for(
+        lambda: not rest_api.collections.policy_profiles.find_by(
+            name=policy_profiles_data[0].get('description')),
+        num_sec=180,
+        delay=10,
+    )
+
+    policy_profiles = [policy_profile.id for policy_profile in rest_api.collections.policy_profiles]
+    rest_api.collections.policy_profiles.delete(policy_profiles)
+    with error.expected("ActiveRecord::RecordNotFound"):
+        rest_api.collections.policy_profiles.action.delete(policy_profiles)
+
+
+def test_edit_policy_profile(policy_profiles_data, rest_api):
+    """Test editing a policy_profile
+
+    Prerequisities:
+        * An appliance with ``/api`` available.
+
+    Steps:
+        * Retrieve list of entities using GET /api/policy_profiles , pick the first one
+        * POST /api/policy_profile/1 (method ``edit``) with the ``description``
+
+    Metadata:
+        test_flag: rest
+    """
+    assert "edit" in rest_api.collections.policy_profiles.action
+
+    try:
+        policy_profile = rest_api.collections.policy_profiles[0]
+    except:
```

Comment:
No, my mistake, it should raise `IndexError` since you will use the array access so you should then check the IndexError and therefore you don't need to do additional checks and you also don't need to do `as e:`. But now you know how to handle the REST API client errors in future :)